server/service/src/sync/translations/invoice.rs

```diff
@@ -178,6 +178,36 @@ pub struct LegacyTransactRow {
     pub om_colour: Option<String>,
 }
 
+/// The mSupply central server will map outbound invoices from omSupply to "si" invoices for the
+/// receiving store.
+/// In the current version of mSupply all om_ fields get copied though.
+/// When receiving the transferred invoice on the omSupply store the "si" get translated to
+/// outbound shipments because the om_type will override to legacy type field.
+/// In other word, the inbound shipment will have an outbound type!
+/// This function detect this case and removes all om_* fields from the incoming record.
+fn sanitize_legacy_record(mut data: serde_json::Value) -> serde_json::Value {
+    let Some(Ok(om_type)) = data
```

Comment:
Hmm, we already have deserialisation from json record to data, shouldn't we use it ? Rather then manually accessing serde fields by value ?

```rust
let data = sanitize_legay_record(serde_json::from_str(&sync_record.data)?);

// ...

fn sanitize_legay_record(data: LegacyTransactType) -> LegacyTransactType {
    if legacy_type == LegacyTransactType::Si && om_type == Some(InvoiceRowType::OutboundShipment) {
        LegacyTransacType {
            om_type: None,
            ..data
        }
    } else {
        data
    }
}
```

Reply:
Or is it because you want to sanitise all of the om_fields dynamically ?

Reply:
Yes I thought about handling all fields case by case but this seemed to become quite messy and was not sure if this causes problem if we would introduce new om_* fields. For this reason I just thought I just remove all om_* fields from the raw json object and leave the rest of the code as it is. So yes its because I remove the fields dynamically.

Reply:
Ok, sounds good, i agree, this dynamic wy is safer for future, and easier to see that it applies to all fields